client/components/state-dropdown/index.js

```diff
@@ -0,0 +1,44 @@
+import React, { PropTypes } from 'react';
+import Dropdown from 'components/dropdown';
+import TextField from 'components/text-field';
+import { translate as __ } from 'lib/mixins/i18n';
+
+const StateDropdown = ( props ) => {
+	const statesMap = ( props.countriesData[ props.countryCode ] || {} ).states;
+
+	if ( ! statesMap ) { // We don't have a list of states for this country
```

Comment:
I mirrored the way WC core builds the states map. For each country:
* If the states map is an associative array, then that's the list of the states for that country.
* If the states map is an **empty** array, then the country has no states.
* If the states map doesn't have an entry for that country, then the country **may** have states, but we don't have a list of them, so show a TextField just in case.